crates/hyperswitch_domain_models/src/payments.rs

```diff
@@ -263,6 +275,153 @@ pub struct PaymentIntent {
     pub routing_algorithm_id: Option<id_type::RoutingId>,
 }
 
+#[cfg(feature = "v2")]
+impl PaymentIntent {
+    fn get_request_incremental_authorization_value(
+        request: &api_models::payments::PaymentsCreateIntentRequest,
+    ) -> common_utils::errors::CustomResult<
+        common_enums::RequestIncrementalAuthorization,
+        errors::api_error_response::ApiErrorResponse,
+    > {
+        request.request_incremental_authorization
+            .map(|request_incremental_authorization| {
+                if request_incremental_authorization == common_enums::RequestIncrementalAuthorization::True {
+                    if request.capture_method == Some(common_enums::CaptureMethod::Automatic) {
+                        Err(errors::api_error_response::ApiErrorResponse::InvalidRequestData { message: "incremental authorization is not supported when capture_method is automatic".to_owned() })?
+                    }
+                    Ok(common_enums::RequestIncrementalAuthorization::True)
+                } else {
+                    Ok(common_enums::RequestIncrementalAuthorization::False)
+                }
+            })
+            .unwrap_or(Ok(common_enums::RequestIncrementalAuthorization::default()))
+    }
+    pub async fn create_domain_model_from_request(
+        state: keymanager::KeyManagerState,
```

Comment:
This function should be use keymanagerState

Reply:
Yes. We are encrypting in this function

Reply:
while converting few fields like shipping and billing address

Reply:
we should not use keymanagerState here, my bad i missed not in the above comment,

We should take shipping and billing as arguments